packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -10,39 +10,148 @@ import 'framework.dart';
 import 'ticker_provider.dart';
 import 'transitions.dart';
 
-class _AnimatedChildSwitcherChildEntry {
-  _AnimatedChildSwitcherChildEntry(this.widget, this.controller, this.animation);
+/// Represents a child that is, or was previously set on the
+/// [AnimatedChildSwitcher.child] field that needs to be laid out using a
+/// [AnimatedChildSwitcherLayoutBuilder].
+abstract class AnimatedChildSwitcherLayoutData {
+  /// The widget that is currently or previously was set as the widget's child,
```

Comment:
Either add the magic null factory constructor here (see the chip pr i submitted) if you want this to be an interface, or add a const constructor with the const constructor boilerplate if you want this to be a superclass (then use "extends" instead of "implements" below).